src/Function.cpp

```diff
@@ -77,6 +77,59 @@ static long cur_func_idx;				// Index into FuncList that we are currently workin
 static bool param_first=true;			// Flag to track output of commas
 static int builtin_functions_cnt;
 
+FunctionAttribute::FunctionAttribute(string name, int prob)
+	:  attribute(name), attribute_probability(prob)
+{
+	//Nothing else to do
+}
+
+void
+FunctionAttribute::OutputAttribute(std::ostream &out, string option)
+{
+	out << " __attribute__((" << attribute << option << "))";
+}
+
+BooleanFunctionAttribute::BooleanFunctionAttribute(string name, int prob)
+	: FunctionAttribute(name, prob)
+{
+	//Nothing else to do
+}
+
+void
+BooleanFunctionAttribute::OutputAttributes(std::ostream &out)
+{
+	if(rnd_flipcoin(attribute_probability))
+		OutputAttribute(out, "");
+}
+
+MultiValuedFunctionAttribute::MultiValuedFunctionAttribute(string name, int prob, vector<string> arguments)
+	: FunctionAttribute(name, prob)
+{
+	attribute_values = arguments;
```

Comment:
Can you please use constructor initializer?

Reply:
Done.